app/main.py

```diff
@@ -1,6 +1,18 @@
-from typing import Callable
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_storage = {}
+
+    def inner(*args, **kwargs) -> Any:
+        key = (args, tuple(sorted(kwargs.items())))
+
+        if key in cache_storage:
+            print("Getting from cache")
+            return cache_storage[key]
+        else:
+            print("Calculating new result")
+            result = func(*args, **kwargs)
+            cache_storage[key] = result
+            return result
```

Comment:
Try to call `return` only once . If `result ` thats a key of your cache storage , you can simply return cache_storage[key] outside the `if ... else` block is not it ?